### ISSUE TITLE
zip vs tar.xz with one vs two readme files

### ISSUE THREAD

**Issue by Kissaki** [ryanoasis/nerd-fonts#1284]
Looking at [v3.0.2](https://github.com/ryanoasis/nerd-fonts/releases/tag/v3.0.2) `CascadiaCode.zip` vs `CascadiaCode.tar.xz`

* the zip has one `readme.md` file
* the tar.xz has a `readme.md` _and_ a `README.md` file

* `README.md`: `# Cascadia Code` information / docs
* `readme.md`: Short `#Nerd Fonts` note

_I assume this is likely an issue with other font packages too, but I only saw and checked on CascadiaCode._

**Comment by Kissaki** [comment]
I would expect them to at least be consistent.

Is there a reason why the packages *should* be different? Other than historic came-to-be?

---

As a release package I guess I would not include too much technical / technical source information.

Given the package is called Cascadia Code despite the patched font being called Caskaya having a generic readme - even short - seems good to at least indicate what the package is [from].

What I like about the more detailed readme is the generic font information. What is the font, where is it from, what is it designed to do / is it designed for. Different font naming reasoning is critical information.

However, `This source font has been preprocessed` is factually wrong.

The `Which font?` section seems very helpful and important.

---

As a simple consistency fix, I would add both files to the zip too.

As a slightly better low-effort fix it could be reasonable to append the extended (source) readme to the generic readme text.  
That'd solve extract file naming conflicts on Windows. And ensure no confusion between having two readme files.

I'm not too familiar with the structure or processes of this project to assess how readme infos could be split better or differently in sources and [produced in] package. But 1. generic Nerd Font info, 2. generic font info, 3. specific font distrib infos (font rename, nerd font naming) would be great.

**Comment by Finii** [comment]
> However, `This source font has been preprocessed` is factually wrong.

What is wrong with that?

**Comment by Finii** [comment]
I think what is missing from both is the release version?
Sure you can see that in the font itself, but maybe better to have it in the/one readme.

When we decouple font updates from patcher releases maybe the 'in parens' version needs to be included also.

**Comment by Finii** [comment]
> That'd solve extract file naming conflicts on Windows.

Baaah, now I get it :woman_facepalming:

**Comment by Finii** [comment]
The Cascadia readme (`README.md` in both archives, no `readme.md` anymore) looks like the following.
Any more ideas?

(The " is factually wrong" from above, I still do not get it, please explain.)

```markdown
# Nerd Fonts

This is an archived font from the Nerd Fonts release 3.0.2.

For more information see:
* https://github.com/ryanoasis/nerd-fonts/
* https://github.com/ryanoasis/nerd-fonts/releases/latest/

# Agave

**Agave** is a succulent monospaced programming font.

https://b.agaric.net/page/agave

For more information have a look at the upstream website: https://github.com/blobject/agave

Version: v37

## Which font?

### TL;DR

* Pick your font family:
  * If you are limited to monospaced fonts (because of your terminal, etc) then pick a font with `Nerd Font Mono` (or `NFM`).
  * If you want to have bigger icons (usually around 1.5 normal letters wide) pick a font without `Mono` i.e. `Nerd Font` (or `NF`). Most terminals support this, but ymmv.
  * If you work in a proportional context (GUI elements or edit a presentation etc) pick a font with `Nerd Font Propo` (or `NFP`).

### Ligatures

Ligatures are generally preserved in the patched fonts.
Nerd Fonts `v2.0.0` had no ligatures in the `Nerd Font Mono` fonts, this has been dropped with `v2.1.0`.
If you have a ligature-aware terminal and don't want ligatures you can (usually) disable them in the terminal settings.

### Explanation

Once you narrow down your font choice of family (`Droid Sans`, `Inconsolata`, etc) and style (`bold`, `italic`, etc) you have 2 main choices:

#### `Option 1: Download already patched font`

 * For a stable version download a font package from the [release page](https://github.com/ryanoasis/nerd-fonts/releases)
 * Or download the development version from the folders here

#### `Option 2: Patch your own font`

 * Patch your own variations with the various options provided by the font patcher (i.e. not include all symbols for smaller font size)

For more information see: [The FAQ](https://github.com/ryanoasis/nerd-fonts/wiki/FAQ-and-Troubleshooting#which-font)

[SIL-RFN]:http://scripts.sil.org/cms/scripts/page.php?item_id=OFL_web_fonts_and_RFNs#14cbfd4a
```

**Comment by Kissaki** [comment]
> > However, `This source font has been preprocessed` is factually wrong.
> 
> What is wrong with that?

I read "this" as referring to the packaged font, in which case labeling it a source font would be wrong. But i see now if you focus on the headlines / context only within the file it makes sense.

The inherent problem is that that readme is a readme of the source (and begins with primarily documenting the source font). A readme in a package should IMO primarily document the package / what the package contains.

**Comment by Kissaki** [comment]
> The Cascadia readme (`README.md` in both archives, no `readme.md` anymore) looks like the following. Any more ideas?

The example you posted is not Cascadia but Agave. I'll interpret it as a generic/neutral example.

I've mentioned it as a simple resolution before, I think this matches it well. I like the version being added too.

The markdown has two level 1 headlines. But I think that's acceptable for the improvement / necessity.

**Comment by Finii** [comment]
> I read "this" as referring to the packaged font,

Ah yes. This is the readme from the source font, just with some added information.
It is hard do formulate that text such that it can life in both places unchanged. _This **source** font_ should give an indicator that it does not mean _this (patched) font_. The patched font is of course always processed (patched), it would not make any sense in that way?

> A readme in a package should IMO primarily document the package / what the package contains. When you open the package and its readme you are interested in the package after all

Yes, but the contents of the package is determined by the source where the fonts has been obtained and which upstream version it is etc. Without that information you still do not know what you have.

> The example you posted is not Cascadia but Agave

Ooops. I wanted Cascadia, because it adds the RFN and renaming clause. But otherwise the process is the same, yes.

> The markdown has two level 1 headlines. But I think that's acceptabl

I believe it is never a problem? Why should you not have multiple level 1 headlines :-D

Will think about some more improvements later. I thought noone will ever read that :grin:

**Comment by Finii** [comment]
![image](https://github.com/ryanoasis/nerd-fonts/assets/16012374/ae4bf544-5236-48ca-a9af-5eac1c8a0ba0)

**Comment by Kissaki** [comment]
> > The markdown has two level 1 headlines. But I think that's acceptabl
> 
> I believe it is never a problem? Why should you not have multiple level 1 headlines :-D

https://github.com/DavidAnson/markdownlint/blob/main/doc/md025.md

> Rationale: A top-level heading is an h1 on the first line of the file, and serves as the title for the document. If this convention is in use, then there can not be more than one title for the document, and the entire document should be contained within this heading.

Although not Markdown specifically, if you take [HTML <h1>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#usage_notes) into account, it should also be used exactly once - as a document heading.

> While using multiple <h1> elements on one page is allowed by the HTML standard (as long as they are not nested), this is not considered a best practice. A page should generally have a single <h1> element that describes the content of the page (similar to the document's <title> element).
